app/assets/javascripts/initializers/initializeReadingListIcons.js

```diff
@@ -29,7 +29,7 @@ function highlightButton(button) {
     button.classList.remove('selected');
   }
   button.addEventListener('click', reactToReadingListButtonClick);
-  button.dataset.clickInitialized = true;
+  button.dataset.saveInitialized = true;
```

Comment:
This fixes the issue for E2E tests, but I know we've done this in a few places now. Maybe not all related to E2E test issues. I get why we're doing it but it adds one more thing to manage when writing code and an E2E test.

After doing some Googling, I came across a project called `cypress-pipe` which looks like it might be the answer to this.

References:

- [When Can The Test Click?](https://www.cypress.io/blog/2019/01/22/when-can-the-test-click)
- [cypress-pipe](https://github.com/NicholasBoll/cypress-pipe)

Reply:
Interesting 🤔  I'm a bit worried that `cypress-pipe` is going to give us false positives in testing though, as it retries clicking on the same element until what you expect to happen happens. I'm thinking particularly in places where we toggle state between open/close, true/false, we could end up passing tests that shouldn't...

Definitely agree it would be good to have some way of handling this in the tests though. Some of the data attributes like this are for app functionality reasons (e.g. not fetching the same data twice), but not all.

Reply:
Yeah, if not `cypress-pipe` maybe something else. We can discuss at the frontend laundry list meeting.